### PR TITLE
Extract necessary functions from `type-fest`

### LINKED ISSUE #85: Refactor to use type-fest
Please see https://github.com/sindresorhus/type-fest/issues/10

### LINKED ISSUE #108: Property 'observable' does not exist on type 'SymbolConstructor'
> Since I found a solution to my problem, this issue is rather informational.

On a small **mono-repo** (`lerna@3.20.2`) project:

- node.js v12.16.1 and npm 6.14.2
- `typescript@^3.8.3` as root `devDependencies`
- `@types/node@^12.12.29` as **root** `devDependencies`
- `@sindre/sorhus@^2.1.0` as **package** `dependencies`

```shell
❯ tsc --build tsconfig.json
node_modules/@sindresorhus/is/dist/index.d.ts:135:13 - error TS2339: Property 'observable' does not exist on type 'SymbolConstructor'.

135     [Symbol.observable](): ObservableLike;
                ~~~~~~~~~~
```

From what I could gather, `observable` on type `SymbolConstructor` is defined in `@types/node@12.12.29` (which is installed in the **root** `node_modules`), but not in `@types/node@13.7.7` (which ends up being installed in the **package's** `node_module` as a side effect.)

One kludgy quick-fix could be to add `@types/node@^12.12.29` in the package's `devDependencies`

Another, arguably cleaner solution is to use `lerna bootstrap --hoist` so that **root** node_modules prevails.

### PR DESCRIPTION
**Pull request by Richienb**
Fixes #85, Fixes #108

// @sindresorhus

### CODE REVIEW

**Comment by sindresorhus** [comment]
I have changed my mind about depending on `type-fest`. I don't think it's worth adding that dependency just for 3 simple types. Can you instead create a types.ts file and copy-paste the types we need from `type-fest` there?

**Comment by Richienb** [comment]
@sindresorhus Fixed!

File: source/index.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@
 /// <reference lib="dom"/>
 /// <reference types="node"/>
 
-export type Class<T = unknown> = new (...args: any[]) => T;
+import {Class, TypedArray, ObservableLike, Primitive} from './types';
```
Comment (sindresorhus) [inline source/index.ts:5]: You need to export these again as they were previously exported to end-users.

Comment (Richienb) [reply source/index.ts:5]: @sindresorhus Fixed!

**Pull request merged**